DOCS.md

```diff
@@ -0,0 +1,205 @@
+| `registry` | `/vela/parameters/npm/registry`, `/vela/secrets/npm/registry` |
+| `email`    | `/vela/parameters/npm/email`, `/vela/secrets/npm/email`       |
+
+Users can use [Vela external secrets](https://go-vela.github.io/docs/concepts/pipeline/secrets/origin/) to substitute these sensitive values at runtime:
+
+```diff
+steps:
+  - name: npm_publish
+    image: target/vela-npm:latest
+    pull: not_present
+    parameters:
+      registry: https://registry.npmjs.org
+-     username: npmUsername
+-     password: superSecretPassword
+```
+
+## Parameters
+
+The following parameters are used to configure the image:
+
+| Name            | Description                                                                                                        | Required | Default                      |
+| --------------- | ------------------------------------------------------------------------------------------------------------------ | -------- | ---------------------------- |
+| `username`      | username for communication with npm                                                                                | `true`   | `N/A`                        |
+| `password`      | password for communication with npm                                                                                | `false`  | `N/A`                        |
+| `email`         | email for communication with npm                                                                                   | `false`  | `N/A`                        |
+| `registry`      | npm instance to communicate with                                                                                   | `false`  | `https://registry.npmjs.org` |
+| `audit_level`   | level at which the audit check should fail (valid options: `low`, `moderate`, `high`, `critical`, `none` to skip)  | `false`  | `low`                        |
+| `strict_ssl`    | whether or not to do SSL key validation during communication                                                       | `false`  | `true`                       |
+| `always_auth`   | force npm to always require authentication                                                                         | `false`  | `false`                      |
+| `skip_ping`     | whether or not to skip `npm ping` authentication command                                                           | `false`  | `false`                      |
+| `dry_run`       | enables pretending to perform the action                                                                           | `false`  | `false`                      |
+| `tag`           | publish package with given alias tag                                                                               | `false`  | `latest`                     |
+| `log_level`     | set the log level for the plugin (valid options: `info`, `debug`, `trace`)                                         | `true`   | `info`                       |
+
```

Comment:
Overall this change look good to me 👍 

However, do you think it's worth adding the `Environment Variables` for the parameters?

Here are some examples for reference:

* https://github.com/go-vela/vela-docker/blob/main/DOCS.md#parameters
* https://github.com/go-vela/vela-git/blob/main/DOCS.md#parameters
* https://github.com/go-vela/vela-hugo/blob/main/DOCS.md#parameters
* https://github.com/go-vela/vela-kaniko/blob/master/DOCS.md#parameters
* https://github.com/go-vela/vela-slack/blob/main/DOCS.md#parameters

Reply:
ah yup.. will do! thanks for the catch

Reply:
fixed.. i know there are more variables we are not mentioning in docs at this time. i'm not entirely sure if that's on purpose. this was mostly transposed from internal docs. 

@Cappo was there a specific reason to not include all the vars usable in the plugin in the docs? if not, i will add the remaining ones.

Reply:
we can add them later too